src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -566,49 +575,138 @@ LRESULT CALLBACK FancyZones::s_WndProc(HWND window, UINT message, WPARAM wparam,
 
 void FancyZones::UpdateZoneWindows() noexcept
 {
-    auto callback = [](HMONITOR monitor, HDC, RECT *, LPARAM data) -> BOOL
     {
+      std::unique_lock writeLock(m_lock);
+      m_zoneWindowMap.clear();
+    }
+
+    struct device
+    {
+      std::wstring _deviceID;
+      HMONITOR _handle;
+
+      std::optional<std::wstring_view> get_generated_part_of_id() const
+      {
+          if (const auto sharpPos = _deviceID.find(L'#'); sharpPos != std::wstring::npos)
+          {
+            return std::wstring_view{_deviceID.data() + sharpPos + 1, size(_deviceID) - sharpPos - 1};
+          }
+          return std::nullopt;
+      }
+    };

```

Comment:
Let's try to follow the exiting coding style, in this case add a new line to separate the code.
Do it also in the rest of the file, to make the code less condensed.